src/request.coffee

```diff
@@ -1,23 +1,19 @@
-npm = require 'npm'
 request = require 'request'
 
 config = require './apm'
 
-loadNpm = (callback) ->
-  npmOptions =
-    userconfig: config.getUserConfigPath()
-    globalconfig: config.getGlobalConfigPath()
-  npm.load(npmOptions, callback)
-
 configureRequest = (requestOptions, callback) ->
-  loadNpm ->
-    requestOptions.proxy ?= npm.config.get('https-proxy') or npm.config.get('proxy') or process.env.HTTPS_PROXY or process.env.HTTP_PROXY
-    requestOptions.strictSSL ?= npm.config.get('strict-ssl')
-
-    userAgent = npm.config.get('user-agent') ? "AtomApm/#{require('../package.json').version}"
-    requestOptions.headers ?= {}
-    requestOptions.headers['User-Agent'] ?= userAgent
-    callback()
+  config.getSetting 'proxy', (proxy) ->
```

Comment:
I think this part is different from before, don't know if this changes anything.

If `getSetting` fails it doesn't call the callback. In the previous code, even if `getSetting` failed, `callback` would still be called. Now `callback` is only called on success.

Don't know anything about the codebase, and the tests pass. Just something to think about. This change is probably better (though it would be nice to handle errors).

(The change of `requestOptions.proxy` would probably go just outside `strict-ssl` if this were strictly-accurate anyways.)